server/src/main/scala/org/http4s/server/ServerBuilder.scala

```diff
@@ -68,14 +69,26 @@ trait SSLSupport { this: ServerBuilder =>
               protocol: String = "TLS",
             trustStore: Option[StoreInfo] = None,
             clientAuth: Boolean = false): Self
+
+}
+
+trait SSLContextSupport { this: ServerBuilder =>
+  def withSSLContext(sslContext: SSLContext, clientAuth: Boolean = false): Self
 }
+
+sealed trait SSLBits
+
 object SSLSupport {
   final case class StoreInfo(path: String, password: String)
-  final case class SSLBits(keyStore: StoreInfo,
-                 keyManagerPassword: String,
-                           protocol: String,
-                         trustStore: Option[StoreInfo],
-                         clientAuth: Boolean)
+  final case class KeyStoreBits(keyStore: StoreInfo,
```

Comment:
Maybe since this is being renamed anyway, we could move it outside SSLSupport object.  That would avoid one more level of import.  Similar for SSLContextBits.  I don't feel strongly, if others do.